lib/jake/string.ex

```diff
@@ -1,21 +1,46 @@
 defmodule Jake.String do
-  def gen(spec) do
-    options = []
-
-    options =
-      if min_length = Map.get(spec, "minLength") do
-        Keyword.put(options, :min_length, min_length)
-      else
-        options
-      end
-
-    options =
-      if max_length = Map.get(spec, "maxLength") do
-        Keyword.put(options, :max_length, max_length)
-      else
-        options
-      end
-
-    StreamData.string(:ascii, options)
+  @strlen_min 1
+
+  @strlen_max 100
+
+  def gen_string(map), do: stringer(map, map["enum"], map["pattern"])
+
+  def find_min_max(map) do
+    min = Map.get(map, "minLength", @strlen_min)
+    max = Map.get(map, "maxLength", @strlen_max)
+    {min, max}
+  end
+
+  def stringer(map, enum, pattern) when is_nil(enum) and is_nil(pattern) do
+    {min, max} = find_min_max(map)
+    re = Randex.stream(~r/[a-zA-Z0-9\_]{#{min},#{max}}/)
+
+    if min <= max do
+      StreamData.bind_filter(StreamData.string(:alphanumeric), fn
+        x when byte_size(x) in min..max ->
+          {:cont, StreamData.constant(x)}
+
+        x when true ->
+          {:cont, Enum.take(re, 1) |> StreamData.member_of()}
+      end)
+    end
+
+    # StreamData.string(:alphanumeric, [{:max_length, max},{:min_length, min}])
+  end
+
+  def stringer(map, enum, pattern) when is_list(enum) do
+    Jake.gen_enum(map["enum"], "string")
+  end
+
+  def stringer(map, enum, pattern) when is_binary(pattern) do
+    {min, max} = find_min_max(map)
+    pat = Randex.stream(~r/#{pattern}/, mod: Randex.Generator.StreamData)
+
+    if min <= max do
+      StreamData.bind_filter(pat, fn
+        x when byte_size(x) in min..max -> {:cont, StreamData.constant(x)}
```

Comment:
what is the different between bind_filter and filter? is bind_filter necessary here?

Reply:
I'll check this out!

Reply:
bind_filter is not necessary here, one can do it with just filter.